File: src/runTests/checkInstall.js
const colors = require('colors');
const commandExists = require('command-exists').sync;
const exec = require("shelljs.exec");
var installed=[];
function checkShell(sh){
  if(commandExists(sh)){
    const reportGood=sh+' is installed!';
    console.log(reportGood.green);
    installed.push(sh);
  }else{
    const reportBad=sh+' is not installed!';
    console.log(reportBad.red);
  };
};

function main(){
  const shells=['dash', 'yash', 'ksh', 'mksh', 'bosh', 'zsh', 'fish', 'bash3', 'bash4', 'bash5', 'heirloom-sh', 'osh', 'bash', 'tcsh'];
  shells.map(x => checkShell(x));
}
main();

File: src/runTests/checkInstall.py
import os
from termcolor import colored


def which(program):
    def is_exe(fpath):
        return os.path.isfile(fpath) and os.access(fpath, os.X_OK)

    fpath, fname = os.path.split(program)
    if fpath:
        if is_exe(program):
            return program
    else:
        for path in os.environ["PATH"].split(os.pathsep):
            exe_file = os.path.join(path, program)
            if is_exe(exe_file):
                return exe_file
    return "Not installed"


def checkShell(program, installed_shells):
    if "/bin" in which(program):
        print(colored(program + " is installed", "green"))
        installed_shells.append(program)
    else:
        print(colored(program + " is not installed", "red"))


if __name__ == "__main__":
    shells = [
        "dash",
        "yash",
        "ksh",
        "mksh",
        "bosh",
        "zsh",
        "fish",
        "bash3",
        "bash4",
        "bash5",
        "heirloom-sh",
        "osh",
        "bash",
        "tcsh",
    ]
    installed = []
    for i in shells:
        checkShell(i, installed)
    # save a list of shells that are installed on the system as 
    # a text file
    f = open("installed_shells", "w")
    for i in installed:
        f.write(i + "\n")
    f.close()
